storage/pruning/pruningStorer.go

```diff
@@ -364,6 +365,77 @@ func (ps *PruningStorer) GetFromEpoch(key []byte, epoch uint32) ([]byte, error)
 
 }
 
+// GetBulkFromEpoch will return a slice of keys only in the persister for the given epoch
+func (ps *PruningStorer) GetBulkFromEpoch(keys [][]byte, epoch uint32) (map[string][]byte, error) {
+	ps.lock.RLock()
+	pd, exists := ps.persistersMapByEpoch[epoch]
+	ps.lock.RUnlock()
+	if !exists {
+		log.Warn("get from removed persister",
+			"id", ps.identifier,
+			"epoch", epoch)
+		return nil, errors.New("persister does not exits")
+	}
+
+	var persisterForEpoch storage.Persister
+	var err error
+
+	if pd.isClosed {
+		persisterForEpoch, err = ps.persisterFactory.Create(pd.path)
+		if err != nil {
+			log.Debug("open old persister", "error", err.Error())
+			return nil, err
+		}
+		defer func() {
+			err = persisterForEpoch.Close()
+			if err != nil {
+				log.Debug("persister.Close()", "error", err.Error())
+			}
+		}()
+		err = persisterForEpoch.Init()
+		if err != nil {
+			log.Debug("init old persister", "error", err.Error())
+			return nil, err
+		}
+	}
+
+	returnMap := make(map[string][]byte, 0)
+	for _, key := range keys {
+		v, ok := ps.cacher.Get(key)
+		if ok {
+			returnMap[string(key)] = v.([]byte)
+			continue
+		}
+
+		if !pd.getIsClosed() {
+			val, err := pd.persister.Get(key)
+			if err != nil {
+				log.Warn("cannot get from active persister",
+					"key", key,
+					"error", err.Error(),
+				)
+				continue
+			}
+
+			returnMap[string(key)] = val
+			continue
+		}
+
+		res, err := persisterForEpoch.Get(key)
```

Comment:
@AdoAdoAdo can you confirm this logic is correct?

- If persister data is closed, then go to the persister and get by key

Reply:
I see the required flow for the entire function something like:
1. Check if the persister for the epoch exists otherwise return an error.
2. If the persister for epoch is closed open it and defer the close so we don't forget
3. Try to read from the cache (we might be lucky)
4. If not lucky read from the persister, which may have been either active, or closed but we opened it.

I guess here we don't need the check pd.getIsClosed, but we can have another variable persisterToRead which can be initialized with either the active persister or the opened persister, and just read from it.

Reply:
done.